### PR TITLE
IntGuesser uses mapping parameter to generate values in given range

### PR DESCRIPTION
**Pull request by mkruk-u2**
Fix for #200, the `IntGuesser` part.

See #202 for `DecimalGuesser` part.

### CODE REVIEW

**Comment by yakobe** [comment]
The build fails due to unrelated issues with doctrine/orm because it's master branch (dev-master) nolonger supports  PHP < 5.6. Also there are some issues with PHP 7.0 but they are "allowed" at the moment.

**Comment by senaria** [comment]
I updated this PR and all checks are running through successfully. The change seems legit. So would you be so nice and merge or at least tell us what speaks against it.
Cheers

File: src/Knp/FriendlyContexts/Guesser/IntGuesser.php [hunk 1]
```diff
@@ -18,7 +18,35 @@ public function transform($str, array $mapping = null)
...
+     * @param array $mapping
+     *
+     * @return int
+     */
+    private function determineMaxValue(array $mapping)
+    {
+        $defaultMax = 2000000000;
+
+        $lengthUndefined = !isset($mapping['length']);
+        if ($lengthUndefined) {
```
Comment (Einenlum) [inline src/Knp/FriendlyContexts/Guesser/IntGuesser.php]: Just a small detail, but why assign a var here instead of doing the test directly in the if?

**Comment by Einenlum** [comment]
:+1:
@Nek- What do you think?

File: src/Knp/FriendlyContexts/Guesser/IntGuesser.php (hunk 1 above)
Comment (mkruk-u2) [inline src/Knp/FriendlyContexts/Guesser/IntGuesser.php]: Just to improve readability, to make the intention clear for the reader.

Comment (Einenlum) [inline src/Knp/FriendlyContexts/Guesser/IntGuesser.php]: :+1:

Comment (Nek-) [inline src/Knp/FriendlyContexts/Guesser/IntGuesser.php]: This looks useless and paraphrases the code...

Comment (yakobe) [inline src/Knp/FriendlyContexts/Guesser/IntGuesser.php]: @Nek- well that was some constructive criticism 😄.
@mkruk-u2 although the variables offer readability, they are not really needed. It looks like the maintainers don't like them, so maybe just inline them?

Comment (Nek-) [inline src/Knp/FriendlyContexts/Guesser/IntGuesser.php]: @yakobe just saying. IMO it's not critical enough to not see the PR merged.

Also I think my notice is pretty accurate. The second condition is not affected by what I said as it actually adds context. (but I still prefer a comment line than an auto-document but useless variable)

nvm let's look into the PR :)

**Comment by Nek-** [comment]
@mkruk-u2 this is great. Can you rebase please ?

**Comment by mkruk-u2** [comment]
@Nek- Would you rather have this variable inlined as suggested, or is it fine as it is now?

**Comment by Nek-** [comment]
@mkruk-u2 fine to me, gonna merge it on my next iteration on the project.

**Pull request merged**